plugins/by-name/ltex-extra/default.nix

```diff
@@ -44,24 +52,65 @@ lib.nixvim.plugins.mkNeovimPlugin {
   };
 
   extraConfig = cfg: {
-    warnings = lib.nixvim.mkWarnings "plugins.ltex-extra" {
-      when = !config.plugins.lsp.enable;
-      message = ''
-        You have enabled `ltex-extra` but not the lsp (`plugins.lsp`).
-        You should set `plugins.lsp.enable = true` to make use of the LTeX_extra plugin's features.
-      '';
-    };
-
-    plugins.lsp = {
-      servers.ltex = {
-        # Enable the ltex language server
-        enable = true;
+    warnings = lib.nixvim.mkWarnings "plugins.ltex-extra" [
+      {
+        when = !lspCfg.enable;
+        message = ''
+          You have enabled `ltex-extra` but not the lsp (`plugins.lsp`).
+          You should set `plugins.lsp.enable = true` to make use of the LTeX_extra plugin's features.
+        '';
+      }
+      (
+        let
+          expectedDefs = map toString [
+            ./.
+            ../../lsp/language-servers
+          ];
+          isExternal = d: !elem d.file expectedDefs;
+          anyExternal =
+            acc: name: v:
+            let
+              e = findFirst isExternal null v.definitionsWithLocations;
+            in
+            if acc != null then
+              acc
+            else if e == null then
+              null
+            else
+              {
+                inherit name;
+                inherit (e) file;
+              };
+          external = foldlAttrs anyExternal null options.plugins.lsp.servers.ltex;
+        in
+        {
+          # TODO: Added 2025-03-30; remove after 25.05
+          # Warn if servers.ltex seems to be configured outside of ltex-extra
+          when = !lspCfg.servers.ltex.enable && external != null;
+          message = ''
+            in ${external.file}
+            You seem to have configured `plugins.lsp.servers.ltex.${external.name}` for `ltex-extra`.
```

Comment:
If we have a reference to the option itself, we can interpolate it here.

```nix
"${options.foo}"
=> "programs.nixvim.foo"
```

options come with a `__toString` (or was it `outPath`?) attr with the value `lib.showOption option.loc`, so interpolating them will stringify to the option path.

This has the advantage of automatically including path prefixes, such as `programs.nixvim`.